### ISSUE TITLE
Bug Report: [Editor Crashes Regardless If File Has Metadata]

### ISSUE THREAD

**Issue by tivvo** [FNF-CNE-Devs/CodenameEngine#347]
### Which target are you using?
[X] Windows
[ ] Linux
[ ] Mac

##### Please check first if your issue haven't already been reported yet.
Editor crashes upon trying to use a song that contains metadata (i.e, artist metada).
_(actual crash error)_
![image](https://github.com/user-attachments/assets/58baad2c-956d-4d35-ae98-57609a4ce283) 

### Recreation Steps
- Use a audio file that contains metadata for an artist (easiest way for this is to get a song from cobalt.tools from youtube, andthen use it after converting it via ffmpeg to ogg)
- Add it into the editor, whether voices or instrumental track, the following should be result:
![tiv@14-07-2024- CodenameEngine #Hind](https://github.com/user-attachments/assets/790af612-369e-4c99-b617-52592e63030a)
**The intended result should be the following:**
![tiv@14-07-2024- CodenameEngine #Ostracod](https://github.com/user-attachments/assets/3020b4b8-9037-48a6-8164-727397a29123)
- After this, create the song as normal and proceed to try to edit the diffculty.
- The game should return an error, and then crash.

**Comment by tivvo** [comment]
this is so stupid lmao

**Comment by tivvo** [comment]
ok I LIEEED this seems to be after encoding with ffmpeg from any file to an ogg
use the following command or whatever to tes tthis out should be obvious what song path is
ffmpeg -i "songPath\song.mp3" -c:a libopus -b:a 128k "songPath\song.ogg"

**Comment by NexIsDumb** [comment]
wait can you pass me the audio
this is so odd😭

**Comment by tivvo** [comment]
ignore that its subterfuge its the quickest thing i could think of that i can reencode for this
[evilterfuge.ogg.zip](https://github.com/user-attachments/files/16755161/evilterfuge.ogg.zip)
also yeah just tested on mac using whisky doesn't wanna work either
![tiv 2024-08-27 @ wine64-preloader@2x](https://github.com/user-attachments/assets/9c16ee04-58cc-4565-bbfb-e9153b2a3d28)

**Comment by NexIsDumb** [comment]
aight, i'll check later on then

**Comment by PetscopF** [comment]
> ### Recreation Steps
> * Use a audio file that contains metadata for an artist (easiest way for this is to get a song from cobalt.tools from youtube, andthen use it after converting it via ffmpeg to ogg)
> * Add it into the editor, whether voices or instrumental track, the following should be result:
>   ![tiv@14-07-2024- CodenameEngine #Hind](https://private-user-images.githubusercontent.com/97667177/348544937-790af612-369e-4c99-b617-52592e63030a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ3MjY3ODIsIm5iZiI6MTcyNDcyNjQ4MiwicGF0aCI6Ii85NzY2NzE3Ny8zNDg1NDQ5MzctNzkwYWY2MTItMzY5ZS00Yzk5LWI2MTctNTI1OTJlNjMwMzBhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODI3VDAyNDEyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk3ZTI5NzBlMzlhMzM4ZjFjNTJhMDFjMGQ0ZTIzNjQyODFlYmE3NWVkYzlkNTQwNDM5YWQ0MWNjNjhlNDhjNTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.AVflxL8cYlQiGD_XhVZdtbCWGd12LdrtwA1B2ugqw_s)
>   **The intended result should be the following:**
>   ![tiv@14-07-2024- CodenameEngine #Ostracod](https://private-user-images.githubusercontent.com/97667177/348544960-3020b4b8-9037-48a6-8164-727397a29123.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ3MjY3ODIsIm5iZiI6MTcyNDcyNjQ4MiwicGF0aCI6Ii85NzY2NzE3Ny8zNDg1NDQ5NjAtMzAyMGI0YjgtOTAzNy00OGE2LTgxNjQtNzI3Mzk3YTI5MTIzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODI3VDAyNDEyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTUzNWU3YmIyYjA1ZDMzYjhjOWNlNjY4YWQ2M2JjNzlhNTM3MGY5OGI5OWEwOWQ5ZWVmNTA4Zjg4NWM0NzExNGYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.4NeDo23-sYDKTvtygTGybRmSB9rnUT_AEnJh2iUTp0c)
> * After this, create the song as normal and proceed to try to edit the diffculty.
> * The game should return an error, and then crash.

I am absolutely confused how implementing an artist into the OGG metadata even causes a Null Object Reference like this. Like, is there just something wrong with how the current audio system implements audio metadata or something that causes it to return a nonexistent object??

**Comment by tivvo** [comment]
> > ### Recreation Steps
> 
> > * Use a audio file that contains metadata for an artist (easiest way for this is to get a song from cobalt.tools from youtube, andthen use it after converting it via ffmpeg to ogg)
> 
> > * Add it into the editor, whether voices or instrumental track, the following should be result:
> 
> >   ![tiv@14-07-2024- CodenameEngine #Hind](https://private-user-images.githubusercontent.com/97667177/348544937-790af612-369e-4c99-b617-52592e63030a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ3MjY3ODIsIm5iZiI6MTcyNDcyNjQ4MiwicGF0aCI6Ii85NzY2NzE3Ny8zNDg1NDQ5MzctNzkwYWY2MTItMzY5ZS00Yzk5LWI2MTctNTI1OTJlNjMwMzBhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODI3VDAyNDEyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk3ZTI5NzBlMzlhMzM4ZjFjNTJhMDFjMGQ0ZTIzNjQyODFlYmE3NWVkYzlkNTQwNDM5YWQ0MWNjNjhlNDhjNTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.AVflxL8cYlQiGD_XhVZdtbCWGd12LdrtwA1B2ugqw_s)
> 
> >   **The intended result should be the following:**
> 
> >   ![tiv@14-07-2024- CodenameEngine #Ostracod](https://private-user-images.githubusercontent.com/97667177/348544960-3020b4b8-9037-48a6-8164-727397a29123.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ3MjY3ODIsIm5iZiI6MTcyNDcyNjQ4MiwicGF0aCI6Ii85NzY2NzE3Ny8zNDg1NDQ5NjAtMzAyMGI0YjgtOTAzNy00OGE2LTgxNjQtNzI3Mzk3YTI5MTIzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODI3VDAyNDEyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTUzNWU3YmIyYjA1ZDMzYjhjOWNlNjY4YWQ2M2JjNzlhNTM3MGY5OGI5OWEwOWQ5ZWVmNTA4Zjg4NWM0NzExNGYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.4NeDo23-sYDKTvtygTGybRmSB9rnUT_AEnJh2iUTp0c)
> 
> > * After this, create the song as normal and proceed to try to edit the diffculty.
> 
> > * The game should return an error, and then crash.
> 
> 
> 
> I am absolutely confused how implementing an artist into the OGG metadata even causes a Null Object Reference like this. Like, is there just something wrong with how the current audio system implements audio metadata or something that causes it to return a nonexistent object??

iirc, even when stripping the metadata still caused an issue, although i do have a theory that the bitrate is too high for the song (it being 128k/b), i'll try a 32k/b encode instead and see if the same thing comes up

wonder if this is a haxel issue also

**Comment by tivvo** [comment]
bad news
![tiv 2024-08-27 @ wine64-preloader@2x](https://github.com/user-attachments/assets/8adcb7ba-309f-4fcc-b0a3-e9c6c07b35f8)
(still fucks up with a lower bitrate)

**Comment by PetscopF** [comment]
I'm guessing that based on the images it's probably somewhere within the audio player.

Does this issue happen with the voices as well?

**Comment by tivvo** [comment]
yep, seems that from the actual crashing error the waveform that generates does not like it at all 
possible solution could be to catch a waveform error and instead not generate it and let the editor just work like normal
(or don't and fix the issue, first thing could save alot of stress though imo)

**Comment by PetscopF** [comment]
My only guess right now is that it searches for the audio data at a very specific location in the file, but the metadata is in place of that and it crashes. I'm not sure if that is correct, I've never delved into how Ogg Vorbis works, I've never delved into how the waveform is generated, I've never delved into how ffmpeg converts into Ogg, but that's really what I've got so far.

**Comment by StrawberrySage** [comment]
> ok I LIEEED this seems to be after encoding with ffmpeg from any file to an ogg use the following command or whatever to tes tthis out should be obvious what song path is ffmpeg -i "songPath\song.mp3" -c:a libopus -b:a 128k "songPath\song.ogg"

You seem to have encoded it using Opus rather than Vorbis here.
I don't think that Flixel natively supports Opus so you need to encode it with Vorbis instead.
Command: `ffmpeg -i "songPath\song.mp3" -c:a libvorbis -b:a 128k "songPath\song.ogg"`

**Comment by MarioGuyAGAIN** [comment]
I CAN HELP! not official dev btw but I could help also

Use cobalt.tools, Much easier way than stinky old ffmpeg.
Hope this helps.

**Comment by r6915ee** [comment]
> Use cobalt.tools, Much easier way than stinky old ffmpeg. Hope this helps.

This is likely only locally. From my understanding, Cobalt only runs on the web.

**Comment by tivvo** [comment]
bro does not know basically everything uses ffmpeg 🥹 i only saw this post now

issue was also stated idk when it'll ge closed but yeah seems to just be a codec issue whatever blah blah

**Comment by r6915ee** [comment]
Yeah; from checking the codec information on several base game songs, Vorbis is very much supported. I believe Opus also includes video capability so that might be why it's not being read properly, but correct me on that if I'm wrong. So I'd probably put it in the Vorbis codec instead, identified in ffmpeg as `libvorbis` (simply `vorbis` may also work; however, ffmpeg identifies it as experimental compared to the former).

**Comment by NexIsDumb** [comment]
btw, we're finally about to fix this
its because the file is an opus which is not supported on lime by default
